network.go

```diff
@@ -0,0 +1,128 @@
+package caddy
+
+import (
+	"errors"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"go.uber.org/zap"
+)
+
+func init() {
+	RegisterModule(ProxyFromEnvironment{})
```

Comment:
These modules prob don't need to be in the project root, can go in modules. Only the interface needs to be here I think.

Reply:
Fair

Reply:
I didn't want to introduce a new package/API surface, so just threw it in `standard`. It also cannot be thrown under `internal`. It's used in `caddyhttp` and `caddytls`. I considered `common`, but it doesn't feel nice. Any ideas?